toogoodtogo_ha_mqtt_bridge/main.py

```diff
@@ -228,10 +276,31 @@ def on_disconnect(client, userdata, rc):
         client.reconnect()
 
 
+def calc_timeout():
+    global watchdog_timeout
+    now = datetime.now()
+    tgtg = settings.get("tgtg")
+
+    if "polling_schedule" not in tgtg:
+        logger.error("No polling_schedule found in settings")
+        os._exit(1)
+
+    cron_schedule = tgtg.polling_schedule
+    if croniter.is_valid(cron_schedule):
+        next_run = croniter(cron_schedule, now).get_next(datetime)
+        for i in range(2):
+            next_run = croniter(cron_schedule, next_run).get_next(datetime)
```

Comment:
Took me a few seconds to realize the intention behind that. If i remember correctly from some comment you try do calculate the next 3 runs?  Good idea! 

But pretty hard to read without any comment. Remember, one day somebody besides you and me will look at that and might be quite confused 😉 

I would not create some additional `cron_schedule` variable. Just use the `tgtg.polling_schedule` here, I see no win by using a slightly  different named var.

I would also use the approach of the itterator mentioned in the croniter doc: https://github.com/kiorky/croniter#about-second-repeats

Last note on that, if you use a `for i in range` loop, but never actually use the i, just call it `_`. see e.g. https://stackoverflow.com/a/5893186/635876

Reply:
Thanks for this Input! 
Fixed with commit [d90e963](https://github.com/MaxWinterstein/toogoodtogo-ha-mqtt-bridge/pull/28/commits/d90e963f49caa8bedf18bb93de67dc0d35af688e)